### PR TITLE
fixed regex logic added tests inc array, linter fix

### PR DESCRIPTION
**Pull request by gityow**
made regex logic more specific , added basic tests and array tests. also added docstring to fill_sql_template.

### CODE REVIEW

File: tests/fugue_sql/test_utils.py [hunk 1]
```diff
@@ -57,3 +63,33 @@ def test_fill_sql_template():
...
+        """select * from tbl where a in ('A','B','C')""" 
+        == fill_sql_template(
+        """select * from tbl where a in (
+            {%- for i in a -%}
+                {%- if loop.index0 < loop.length - 1 -%}'{{-i|upper-}}',
+                {%- else -%}'{{-i|upper-}}'
+                {%- endif -%}
+            {%- endfor -%}
+            )""", data)
+    )
```
Comment (goodwanghan) [inline tests/fugue_sql/test_utils.py:95]: i think you need to add one blank line here

File: tests/fugue_sql/test_utils.py [hunk 2]
```diff
@@ -3,6 +3,12 @@
 
 def test_fill_sql_template():
     data = {"a": 1, "b": "x"}
+    assert ( "select * from tbl where a = 1 and b = 'x'" 
+    == fill_sql_template("select * from tbl where a = {{a}} and b = '{{b}}'", data)
+    )
+    assert ("""select * from tbl where a = 1 and b = "x" """ 
+    == fill_sql_template("""select * from tbl where a = {{a}} and b = "{{b}}" """, data)
+    )
```
Comment (goodwanghan) [inline tests/fugue_sql/test_utils.py:11]: what about a case
`"%{{b}}"`

will this be handled correctly?

Comment (goodwanghan) [reply tests/fugue_sql/test_utils.py:11]: `select * where b="%{{b}}"`

**Comment by gityow** [comment]
thanks for catching another good case. I found more cases:
```
"{}%{}" == "{}%{}"
"{a}%{b}" == ???
"{{b}}%" == "x%"
"{}%{}" == "{}%{}"
"%}" == "}%"
```

**Comment by gityow** [comment]
the closest I got was to change to this but it fails some tests beginning with \ char.
`MATCH_QUOTED_STRING = r"([\"'])(({|%|})*)\1"`

**Comment by gityow** [comment]
removed test cases beginning with `\` such as `\"{%'{%'\""` and advise users to add `{% raw %} {% endraw %}` block around sql that is jinja template sensitive

**Review by goodwanghan — APPROVED** [review]

**Pull request merged**